pkg/agents/team6/consumption.go

```diff
@@ -77,17 +78,83 @@ func FoodRequired(currentHP int, goalHP int, healthInfo *health.HealthInfo) food
 	return food.FoodType(healthInfo.Tau * math.Log(healthInfo.Width/denom))
 }
 
+func (a *CustomAgent6) incomingTreatyValidityAndFoodRange(trty messages.Treaty) (int, int, bool) {
```

Comment:
Can you split this into 2 functions? Sorry to be a pain - they just do quite different things (1 iterates through all treaties, the other sees if only 1 is valid). Surely foodRange doesn't need any arguments?

Reply:
foodRange will need the list of active treaties

Reply:
yeah exactly. This is innate to the agent (so doesn't require arguments).

Reply:
true, but it doesn't need the incoming treaty

Reply:
just call food range inside of the valid treaty function. If I want the food range, I shouldn't have to always pass in a new treaty